### PR TITLE
eth_getStorageAt accept second argument (location) with leading zeros…

### PR DESCRIPTION
**Pull request by NazariiDenha**
… too

Context: https://app.asana.com/0/1202293017617135/1205861222088883/f

This pr adds new type which is the same as hexutil.Big, but also accept numbers with leading zeros

### CODE REVIEW

File: rpc/cfx_api.go [hunk 1]
```diff
@@ -115,10 +116,10 @@ func (api *cfxAPI) GetCode(ctx context.Context, contract types.Address, epoch *t
 	return cfx.GetCode(contract, toSlice(epoch)...)
 }
 
-func (api *cfxAPI) GetStorageAt(ctx context.Context, address types.Address, position *hexutil.Big, epoch *types.Epoch) (hexutil.Bytes, error) {
+func (api *cfxAPI) GetStorageAt(ctx context.Context, address types.Address, location *rpcgtypes.Location, epoch *types.Epoch) (hexutil.Bytes, error) {
```
Comment (Thegaram) [inline rpc/cfx_api.go]: This seems to be an overkill. Geth uses `common.Hash` and works well with both types of calls. Can we just use `common.Hash`?

Comment (NazariiDenha) [reply rpc/cfx_api.go]: `common.Hash` checks during unmarhsalling that the input is of length exactly 32. And I tested it locally, it works this way. I didn't find any time that is suitable here

Comment (Thegaram) [reply rpc/cfx_api.go]: Okay, maybe this is the entry point for geth's implementation then, can we use the same approach?

https://github.com/scroll-tech/go-ethereum/blob/a8a9dc83cb697f5ecb764acab8ef93ffa6f66379/internal/ethapi/api.go#L833-L843

Comment (NazariiDenha) [reply rpc/cfx_api.go]: yes, nice point, changed to string

File: rpc/cfx_api.go [hunk 2]
```diff
@@ -115,10 +116,10 @@ func (api *cfxAPI) GetCode(ctx context.Context, contract types.Address, epoch *t
 	return cfx.GetCode(contract, toSlice(epoch)...)
 }
 
-func (api *cfxAPI) GetStorageAt(ctx context.Context, address types.Address, position *hexutil.Big, epoch *types.Epoch) (hexutil.Bytes, error) {
+func (api *cfxAPI) GetStorageAt(ctx context.Context, address types.Address, key string, epoch *types.Epoch) (hexutil.Bytes, error) {
```
Comment (Thegaram) [inline rpc/cfx_api.go]: No need to modify `cfxAPI`.

**Review by Thegaram — APPROVED** [review]
Looks good. We can merge it after you've tested it.

File: rpc/cfx_api.go (hunk 2 above)
Comment (NazariiDenha) [reply rpc/cfx_api.go]: only eth api?

**Comment by NazariiDenha** [comment]
removed change of `cfx_api`
locally tested, it accepts the same args as other nodes can

**Review by Thegaram — APPROVED** [review]

**Pull request merged**